game.py

```diff
@@ -20,7 +20,38 @@ def snowman(snowman_word):
     If the player wins and, 
     'Sorry, you lose! The word was {snowman_word}' if the player loses
     """
-    pass
+    
+    snowman_word_dic = build_letter_status_dict(snowman_word)
+    correct_letter_guess_statuses = build_letter_status_dict(snowman_word)
+    wrong_guesses_count = 0
+    wrong_guesses_list = set()
+
+    while wrong_guesses_count < SNOWMAN_MAX_WRONG_GUESSES:
+        print_snowman_graphic(wrong_guesses_count)
+        print_word_progress_string(snowman_word, correct_letter_guess_statuses)
+
+        valid_character = get_letter_from_user(snowman_word_dic, wrong_guesses_list)
```

Comment:
This is where the duplicate dictionaries from above start to cause an issue. You pass `snowman_word_dic` to the `get_letter_from_user` function, but when you find a letter in the word, you update the `correct_letter_guess_statuses`. This actually causes an issue where the `get_letter_from_user` function doesn't register when a correct letter is guessed a second time. During play testing, this means that I'm never notified if I am picking a letter that has already been discovered. While this will pass all the tests, it does mean the program doesn't quite behave as expected.